### PR TITLE
Add coverage reports check in CI

### PR DESCRIPTION
**Pull request by ClementWalter**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

The `make test` command is run in the CI but it's coverage output is not used to block PR, nor at least recorded.

Issue Number: #280 

## What is the new behavior?

Use codecov to display coverage in PR and enforce minimal coverage in CI

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -73,3 +73,7 @@ jobs:
       - name: Run tests
         run: |
           make test-no-log
+      - name: Upload coverage report to codecov
+        uses: codecov/codecov-action@v3
+        with:
+          files: ./coverage/coverage.json
```
Comment (Eikix) [inline .github/workflows/test.yml:79]: do we need additional steps to attach codecov integration in our PRs?
something like https://github.com/apps/codecov

**Review by Eikix — APPROVED** [review]
lgtm

**Review by Eikix — COMMENTED** [review]
does it work now?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -34,11 +34,7 @@ jobs:
         if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
         run: make setup
       - name: Compile all the cairo files
-        run: |
-          make format-check
-          for filename in $(find ./src/ -type f -name '*.cairo');
-          do echo $(basename -- $filename) && starknet-compile $filename --cairo_path ./src --disable_hint_validation >/dev/null;
-          done;
+        run: make build
```
Comment (Eikix) [inline .github/workflows/test.yml:37]: do we stop checking for formatting?

**Review by Eikix — DISMISSED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (ClementWalter) [reply .github/workflows/test.yml:37]: good catch I forgot it

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**